assets/src/edit-story/components/form/numeric.js

```diff
@@ -160,42 +209,80 @@ function Numeric({
         value={
           isMultiple
             ? ''
-            : `${inputValue}${dot ? DECIMAL_POINT : ''}${focused ? '' : symbol}`
+            : `${inputValue.length > 0 ? inputValue : value}${
+                focused ? '' : symbol
```

Comment:
I noticed when typing before almost every time a character was typed the cursor would skip to the end of the input.  Wasn't really notice-able when typing from scratch, but modifying a number it was super apparent.

I was able to fix this by largely making `value` the source of truth.  Whenever we type something, `onChange` is called and the `value` that returns is what gets displayed.  However, for those moments that we have just a negative symbol ('-') or a bunch of zeroes, or some other input we don't yet want to parse, those values get stored locally in `inputValue`.  And once the input is parseable, then we parse it, send it down the `onChange` pipeline, reset `inputValue`, and then `value` returns to being the source of truth.